driven/storage/adapter.go

```diff
@@ -379,3 +379,86 @@ func (m *database) onDataChanged(changeDoc map[string]interface{}) {
 		m.listener.OnCollectionUpdated(coll.(string), recordMap)
 	}
 }
+
+// Get Survey
+func (sa *Adapter) GetSurvey(id string) (*model.Survey, error) {
+
+	filter := bson.M{"_id": id}
+	var list []model.Survey
+	err := sa.db.surveys.Find(filter, &list, &options.FindOptions{})
+	if err != nil {
+		fmt.Printf("error storage.Adapter.GetSurvey(%s) - %s", id, err)
+		return nil, fmt.Errorf("error storage.Adapter.GetSurvey(%s) - %s", id, err)
+	}
+
+	if len(list) > 0 {
+		entry := list[0]
+		return &entry, nil
+	} else {
+		return nil, nil
+	}
+}
+
+// Create Survey
+func (sa *Adapter) CreateSurvey(survey model.Survey) (*model.Survey, error) {
+
+	_, err := sa.db.surveys.InsertOne(survey)
+	if err != nil {
+		fmt.Printf("error storage.Adapter.CreateSurvey(%s) - %s", survey.ID, err)
+		return nil, fmt.Errorf("error storage.Adapter.CreateSurvey(%s) - %s", survey.ID, err)
+	}
+
+	return &survey, nil
+}
+
+// Update Survey
+func (sa *Adapter) UpdateSurvey(user *model.User, survey model.Survey) (*model.Survey, error) {
+
+	if len(survey.ID) > 0 {
+
+		now := time.Now().UTC()
+		filter := bson.M{"_id": survey.ID, "creator_id": user.Claims.Subject}
+		update := bson.M{"$set": bson.M{
+			"org_id":       survey.OrgID,
+			"app_id":       survey.AppID,
+			"questions":    survey.Questions,
+			"scored":       survey.Scored,
+			"result_rule":  survey.ResultRule,
+			"type":         survey.Type,
+			"stats":        survey.SurveyStats,
+			"sensitive":    survey.Sensitive,
+			"date_updated": now,
+		}}
+
+		result, err := sa.db.surveys.UpdateOne(filter, update, nil)
+		if err != nil {
+			fmt.Printf("error storage.Adapter.UpdatePoll(%s) - %s", survey.ID, err)
+			return nil, fmt.Errorf("error storage.Adapter.UpdatePoll(%s) - %s", survey.ID, err)
+		}
+
+		modifiedCount := result.ModifiedCount
+		if modifiedCount == 0 {
+			return nil, fmt.Errorf("error storage.Adapter.DeleteSurvey(): 403 - Not the creator ", err)
+		}
+	}
+
+	return &survey, nil
```

Comment:
There should be no need to return the survey since we are just applying the changes that the client sent. The update API can simply return a "success" message if no error is returned here. If we did want to return this data, we would need to use the `FindAndUpdateOne` storage function to get the latest data from the database to ensure we are not accidentally allowing the user to set some values in the response (eg. `creator_id`).